File: frontend/src/api/persona.service.ts
// Capa de API de Personas.
import http from '@/api/http'
import type { PersonaDetalleDto, PersonaDto } from '@/types/persona.type'

export function listarPersonas() {
  return http.get<PersonaDetalleDto[]>('/personas/listar').then((r) => r.data)
}

export function crearPersona(dto: PersonaDto) {
  return http.post<PersonaDetalleDto>('/personas/crear', dto).then((r) => r.data)
}

export function actualizarPersona(idPersona: number, dto: PersonaDto) {
  return http
    .put<PersonaDetalleDto>('/personas/actualizar', dto, { params: { idPersona } })
    .then((r) => r.data)
}

export function eliminarPersona(idPersona: number) {
  return http.delete('/personas/eliminar', { params: { idPersona } })
}

File: src/main/java/com/uap/control_tickets/services/interfaces/PersonaService.java
package com.uap.control_tickets.services.interfaces;

import com.uap.control_tickets.dto.persona.PersonaDetalleDto;
import com.uap.control_tickets.dto.persona.PersonaDto;

import java.util.List;

/** Contrato del CRUD de Persona. */
public interface PersonaService {
    List<PersonaDetalleDto> listar();
    PersonaDetalleDto obtener(Long idPersona);
    PersonaDetalleDto crear(PersonaDto dto);
    PersonaDetalleDto actualizar(Long idPersona, PersonaDto dto);
    void eliminar(Long idPersona);
}
